scripts/e2e-deploy.sh

```diff
@@ -121,19 +109,36 @@ echo "Adding registry.redhat.io secret."
 # the json distributed by access.redhat.com/terms-based-registry is a nested object.
 # oc wants the contents of the .data object, so we need to unwrap the outer layer
 # in order to load the pull secrets dockerconfigjson object into the secret.
-if [ -f ${REGISTRY_REDHAT_IO_SECRETS} ]; then
-    SECRET=$(cat ${REGISTRY_REDHAT_IO_SECRETS} | \
+if [ -f $REGISTRY_REDHAT_IO_SECRETS ]; then
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        # Mac OSX
+        SECRET=$(cat $REGISTRY_REDHAT_IO_SECRETS | \
+             python -c 'import yaml, sys; print(yaml.safe_load(sys.stdin).get("data").get(".dockerconfigjson"))' | \
+             base64 -D)
+    else
+            SECRET=$(cat $REGISTRY_REDHAT_IO_SECRETS | \
              python -c 'import yaml, sys; print(yaml.safe_load(sys.stdin).get("data").get(".dockerconfigjson"))' | \
              base64 -d)
+    fi
     # we need to install the pull secret into multiple projects because setting
     # up a shared secret across projects is not well-supported by OCP <=4.2.
-    for project in "${BUILDFACTORY_PROJECT}" "${DEPLOY_PROJECT}"; do
-        echo ${SECRET} | ${OC} create secret generic registry-redhat-io-secret \
+    for project in "${BUILDFACTORY_PROJECT}" "${DEPLOY_PROJECT}" "${SECRETS_PROJECT}"; do
+        echo ${SECRET} | ${OC} create secret generic rh-registry-pull-secret \
+                                    --from-file=.dockerconfigjson=/dev/stdin \
+                                    -n ${project} \
+                                    --type=kubernetes.io/dockerconfigjson
+        ${OC} secrets link default rh-registry-pull-secret -n ${project} --for=pull
+        ${OC} secrets link builder rh-registry-pull-secret -n ${project}
+    done
+
+    # i'm not sure what the quay secret should be so i copied the above secret and just named it quay-cloudservices-push
+    for project in "${SECRETS_PROJECT}"; do
+        echo ${SECRET} | ${OC} create secret generic quay-cloudservices-push \
```

Comment:
My biggest question/concern is that I don't know what this secret (`quay-cloudservices-push`) should be set to. It was required so i set it to the same value as the `rh-registry-pull-secret` during debugging and it worked for me - but I would rather it be what it is supposed to be. If anyone has any ideas - let me know 😄

Reply:
@abaiken  My guess is that this is related to the two very new PRs that @bsquizz just added to the e2e-deploy repo:
https://github.com/RedHatInsights/e2e-deploy/pull/971
https://github.com/RedHatInsights/e2e-deploy/pull/990

Looks like its just another private registry secret based on the sanitized secret here:
https://github.com/RedHatInsights/e2e-deploy/blob/5e81236147a5064bf83b1df19d7a15da0b31b903/secrets/sanitized/quay-cloudservices-push.yml

I think what you have is okay for now, but we could follow up with Brandon for more clarity. Maybe beneficial for that repo to add some documentation about it so all teams have some more clarity.

Reply:
Ah! It looks like it is literally expecting the yaml - I will update the script to add the yaml file instead of unwrapping it to get the contents of the nested object. Thanks @chambridge

Reply:
Based on what @bsquizz said on Friday, I think the tl;dr is that there's a secrets yaml file that they're side-loading to fill in the secrets. So, we just need to decide where we want to keep our secrets and do a similar thing.

Reply:
https://github.com/project-koku/koku/issues/1644 I filed an issue to use the new format 👍  But this was about the script requiring a secret that we actually don't use. I reached out to Brandon about it and he said he was going to remove it from the buildfactory config so we will no longer have to create it.